crates/sp-domains/src/fraud_proof.rs

```diff
@@ -199,23 +200,47 @@ impl FalseInvalidBundleEntryFraudProof {
     }
 }
 
-// TODO: Define rest of the fraud proof fields
+/// Proof data specific to each *expected* invalid bundle type
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub enum ProofDataPerExpectedInvalidBundle {
+    OutOfRangeTx {},
+}
+
 #[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
 pub struct TrueInvalidBundleEntryFraudProof {
+    bad_receipt_hash: ReceiptHash,
     domain_id: DomainId,
     bundle_index: u32,
+    mismatched_extrinsic: OpaqueExtrinsic,
+    mismatched_extrinsic_index: u32,
+    extrinsic_inclusion_proof: Vec<Vec<u8>>,
+    proof_data: ProofDataPerExpectedInvalidBundle,
 }
 
 impl TrueInvalidBundleEntryFraudProof {
-    pub fn new(domain_id: DomainId, bundle_index: u32) -> Self {
+    pub fn new(
+        bad_receipt_hash: ReceiptHash,
+        domain_id: DomainId,
+        bundle_index: u32,
+        mismatched_extrinsic: OpaqueExtrinsic,
+        mismatched_extrinsic_index: u32,
+        extrinsic_inclusion_proof: Vec<Vec<u8>>,
```

Comment:
thinking on this, we are passing extrinsic data which is also available in the inclusion proof so does the duplication here necessary ?
cc @NingLin-P

Reply:
I am not sure I get you. I have removed encoded extrinsic from fields as part of optimization later on. Let me rework the commits.